com.unity.multiplayer.mlapi/Runtime/NetworkVariable/Collections/NetworkSet.cs

```diff
@@ -478,8 +455,7 @@ public void UnionWith(IEnumerable<T> other)
             }
         }
 
-        /// <inheritdoc />
-        bool ISet<T>.Add(T item)
+        public void Add(T item)
```

Comment:
nit: You don't need 3 methods for this. Just put the implementation in the ISet and call it from the ICollection.

Reply:
I think the wackiness here is that we have `ISet<T>` which has its own `Add` method, and then `ICollection<T>` (which is a parent to `ISet<T>`) also has its own `Add` method, which differs only by return type, and so I have to implement both explicit interfaces.

But then if a user wants to call either one, they have to do a  non-intuitive cast, e.g. `((ISet<int>)m_ServerComp.TheSet).Add(1234);`.  That's why I did the 3rd method.  But maybe I'm missing another way?

I'm thinking about using a less grand interface than ISet / ICollection anyway vs. today where we promise to do everything those interfaces do, so that if we use native container underlying storage we don't have to make up for make of these calls that the native types don't support.

Reply:
`HashSet<T>` is similar and it implements both `ISet<T>` and `ICollection<T>` at the same time — and yes, it implements both `void` returning and `bool` returning methods and they both call the exact same method behind the scenes. Let me get you some useful links to relevant points.

`bool Add`:
```cs
public bool Add(T item) => AddIfNotPresent(item, out _);
```
https://github.com/dotnet/runtime/blob/1a4675db08384e4f20cf4481e6eaf03e40f3d754/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs#L455

`void Add`:
```cs
void ICollection<T>.Add(T item) => AddIfNotPresent(item, out _);
```
https://github.com/dotnet/runtime/blob/1a4675db08384e4f20cf4481e6eaf03e40f3d754/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs#L181

underlying `AddIfNotPresent` (the actual code):
```cs
private bool AddIfNotPresent(T value, out int location)
{
    if (_buckets == null)
    {
        Initialize(0);
    }
    Debug.Assert(_buckets != null);

    Entry[]? entries = _entries;
// ...
```
https://github.com/dotnet/runtime/blob/1a4675db08384e4f20cf4481e6eaf03e40f3d754/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs#L1084

I think we could follow the same interface pattern closely to make our stuff more .NET-like :)

Reply:
Yeah, I ended up sending Matt much the same thing in a DM & [gist](https://gist.github.com/JesseOlmer/3e862f0a6932f66cd4d7b0aebaeb4d9e). The key here is you don't need explicit implementation on both of them, just enough to disambiguate. Explicit implementation of both is also what causes this required cast in the first place, as explicit implementations always need an explicit cast.